### PR TITLE
Add function test

### PR DESCRIPTION
**Pull request by Zherphy**
增加部分单元测试

### CODE REVIEW

**Comment by shishupei** [comment]
开始门禁检查，等门禁检查通过后可以合入

**Comment by shishupei** [comment]
开始门禁检查，等门禁检查通过后可以合入

**Comment by shishupei** [comment]
<table style="border-collapse: collapse" border=1>
    <tr>
        <th>检查项</th>
        <th>状态</th>
        <th>日志</th>
    </tr>

<tr>
    <td>Check代码检查</td>
    <td>&#10060;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_Check代码检查.html">查看日志</a></td>
</tr>

<tr>
    <td>DT覆盖率</td>
    <td>42.3
%</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_测试用例覆盖率检查.html">查看日志</a></td>
</tr>

<tr>
    <td>敏感信息扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_敏感信息扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>开源代码漏洞扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_开源代码漏洞扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>安全编码扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_安全编码扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>流水线链接</td>
    <td colspan="2"><a href="https://devcloud.cn-north-4.huaweicloud.com/cicd/project/3a76c1785dda4b13a399937d1978f240/pipeline/detail/58d2a62564c14ce1815dcce4d1064020/280a129c686d4e04b8c46ae13f2b0792">点击跳转</a></td>
</tr>
</table>

**Comment by shishupei** [comment]
<table style="border-collapse: collapse" border=1>
    <tr>
        <th>检查项</th>
        <th>状态</th>
        <th>日志</th>
    </tr>

<tr>
    <td>Check代码检查</td>
    <td>&#10060;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_Check代码检查.html">查看日志</a></td>
</tr>

<tr>
    <td>DT覆盖率</td>
    <td>42.3
%</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_测试用例覆盖率检查.html">查看日志</a></td>
</tr>

<tr>
    <td>敏感信息扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_敏感信息扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>开源代码漏洞扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_开源代码漏洞扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>安全编码扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_安全编码扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>流水线链接</td>
    <td colspan="2"><a href="https://devcloud.cn-north-4.huaweicloud.com/cicd/project/3a76c1785dda4b13a399937d1978f240/pipeline/detail/58d2a62564c14ce1815dcce4d1064020/e35cc1b66e654a0ba7b21f492f6c4daa">点击跳转</a></td>
</tr>
</table>

**Comment by shishupei** [comment]
开始门禁检查，等门禁检查通过后可以合入

**Comment by shishupei** [comment]
开始门禁检查，等门禁检查通过后可以合入

**Comment by shishupei** [comment]
<table style="border-collapse: collapse" border=1>
    <tr>
        <th>检查项</th>
        <th>状态</th>
        <th>日志</th>
    </tr>

<tr>
    <td>Check代码检查</td>
    <td>&#10060;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_Check代码检查.html">查看日志</a></td>
</tr>

<tr>
    <td>DT覆盖率</td>
    <td>46.2
%</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_测试用例覆盖率检查.html">查看日志</a></td>
</tr>

<tr>
    <td>敏感信息扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_敏感信息扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>开源代码漏洞扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_开源代码漏洞扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>安全编码扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_安全编码扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>流水线链接</td>
    <td colspan="2"><a href="https://devcloud.cn-north-4.huaweicloud.com/cicd/project/3a76c1785dda4b13a399937d1978f240/pipeline/detail/58d2a62564c14ce1815dcce4d1064020/97f771332017464fbfbab6b49e453568">点击跳转</a></td>
</tr>
</table>

**Comment by shishupei** [comment]
开始门禁检查，等门禁检查通过后可以合入

**Comment by shishupei** [comment]
<table style="border-collapse: collapse" border=1>
    <tr>
        <th>检查项</th>
        <th>状态</th>
        <th>日志</th>
    </tr>

<tr>
    <td>Check代码检查</td>
    <td>&#10060;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_Check代码检查.html">查看日志</a></td>
</tr>

<tr>
    <td>DT覆盖率</td>
    <td>46.2
%</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_测试用例覆盖率检查.html">查看日志</a></td>
</tr>

<tr>
    <td>敏感信息扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_敏感信息扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>开源代码漏洞扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_开源代码漏洞扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>安全编码扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_安全编码扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>流水线链接</td>
    <td colspan="2"><a href="https://devcloud.cn-north-4.huaweicloud.com/cicd/project/3a76c1785dda4b13a399937d1978f240/pipeline/detail/58d2a62564c14ce1815dcce4d1064020/21260f3a4ee549beadab9295e444ada5">点击跳转</a></td>
</tr>
</table>

**Comment by shishupei** [comment]
开始门禁检查，等门禁检查通过后可以合入

**Comment by shishupei** [comment]
开始门禁检查，等门禁检查通过后可以合入

**Comment by shishupei** [comment]
<table style="border-collapse: collapse" border=1>
    <tr>
        <th>检查项</th>
        <th>状态</th>
        <th>日志</th>
    </tr>

<tr>
    <td>Check代码检查</td>
    <td>&#10060;</td>
    <td>任务失败, 请重试</td>
</tr>

<tr>
    <td>DT覆盖率</td>
    <td>47.7
%</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_测试用例覆盖率检查.html">查看日志</a></td>
</tr>

<tr>
    <td>敏感信息扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_敏感信息扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>开源代码漏洞扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_开源代码漏洞扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>安全编码扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_安全编码扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>流水线链接</td>
    <td colspan="2"><a href="https://devcloud.cn-north-4.huaweicloud.com/cicd/project/3a76c1785dda4b13a399937d1978f240/pipeline/detail/58d2a62564c14ce1815dcce4d1064020/ecd695189cdd43d588783496819cb2f4">点击跳转</a></td>
</tr>
</table>

**Comment by shishupei** [comment]
<table style="border-collapse: collapse" border=1>
    <tr>
        <th>检查项</th>
        <th>状态</th>
        <th>日志</th>
    </tr>

<tr>
    <td>Check代码检查</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_Check代码检查.html">查看日志</a></td>
</tr>

<tr>
    <td>DT覆盖率</td>
    <td>47.7
%</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_测试用例覆盖率检查.html">查看日志</a></td>
</tr>

<tr>
    <td>敏感信息扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_敏感信息扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>开源代码漏洞扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_开源代码漏洞扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>安全编码扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_安全编码扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>流水线链接</td>
    <td colspan="2"><a href="https://devcloud.cn-north-4.huaweicloud.com/cicd/project/3a76c1785dda4b13a399937d1978f240/pipeline/detail/58d2a62564c14ce1815dcce4d1064020/496a18c2f1a64f1fae32ef9dd843dd43">点击跳转</a></td>
</tr>
</table>

File: server/server_test.go [hunk 1]
```diff
@@ -0,0 +1,652 @@
...
+	ttl:          time.Hour,
+	bucket:       "Bucket",
+	prefix:       "Prefix",
+	cdnDomain:    "CDNDomain",
+	isAuthorized: auth.GiteeAuth(),
+}
+
+const unexpectedPanic = "unexpected panic value or wantErr mismatch"
+const expectedPanic = "expected panic but none occurred"
+const batchUrlPath = "/owner/repo/objects/batch"
```
Comment (wj00037) [inline server/server_test.go:41]: 多个const并列时组合到一个。若无逻辑关系按从短到长排列
const (
 batchUrlPath       = "/owner/repo/objects/batch"
 expectedPanic     = "expected panic but none occurred"
 unexpectedPanic = "unexpected panic value or wantErr mismatch"
)

**Comment by shishupei** [comment]
<table style="border-collapse: collapse" border=1>
    <tr>
        <th>检查项</th>
        <th>状态</th>
        <th>日志</th>
    </tr>

<tr>
    <td>Check代码检查</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_Check代码检查.html">查看日志</a></td>
</tr>

<tr>
    <td>DT覆盖率</td>
    <td>47.7
%</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_测试用例覆盖率检查.html">查看日志</a></td>
</tr>

<tr>
    <td>敏感信息扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_敏感信息扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>开源代码漏洞扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_开源代码漏洞扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>安全编码扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_安全编码扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>流水线链接</td>
    <td colspan="2"><a href="https://devcloud.cn-north-4.huaweicloud.com/cicd/project/3a76c1785dda4b13a399937d1978f240/pipeline/detail/58d2a62564c14ce1815dcce4d1064020/6c16138863434a36aebf1a1072b25f8d">点击跳转</a></td>
</tr>
</table>

**Comment by shishupei** [comment]
开始门禁检查，等门禁检查通过后可以合入

**Comment by shishupei** [comment]
<table style="border-collapse: collapse" border=1>
    <tr>
        <th>检查项</th>
        <th>状态</th>
        <th>日志</th>
    </tr>

<tr>
    <td>Check代码检查</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_Check代码检查.html">查看日志</a></td>
</tr>

<tr>
    <td>DT覆盖率</td>
    <td>47.7
%</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_测试用例覆盖率检查.html">查看日志</a></td>
</tr>

<tr>
    <td>敏感信息扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_敏感信息扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>开源代码漏洞扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_开源代码漏洞扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>安全编码扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/BigFiles/42/42_安全编码扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>流水线链接</td>
    <td colspan="2"><a href="https://devcloud.cn-north-4.huaweicloud.com/cicd/project/3a76c1785dda4b13a399937d1978f240/pipeline/detail/58d2a62564c14ce1815dcce4d1064020/ef2e7dfcecbe47d7b5ebeed241c8473f">点击跳转</a></td>
</tr>
</table>

**Review by wj00037 — APPROVED** [review]

**Pull request merged**